app/users/serializers/AdminSerializer.py

```diff
@@ -103,17 +104,34 @@ def to_representation(self, instance):
         Customize the representation of an Admin instance.
 
         This method customizes the serialized representation of an Admin instance,
-          including handling superuser rights.
+        including handling superuser rights.
 
         Args:
             instance (Admin): The admin instance to represent.
 
         Returns:
             dict: The customized serialized representation of the admin instance.
         """
+        CODENAME_TO_RIGHTS = {
+            codename: right
+            for right, data in PERMISSION_CODENAMES.items()
+            for codename in data["permissions"]
+        }
         data = super().to_representation(instance)
         if instance.super_user:
-            data["rights"] = ["__All__"]
+            data["rights"] = [
+                {"right": right, "color": PERMISSION_CODENAMES[right]["color"]}
+                for right in PERMISSION_CODENAMES
+            ]
         else:
-            data["rights"] = list(instance.permissions.values_list("codename", flat=True))
+            permissions = instance.permissions.values_list("codename", flat=True)
+            rights_set = set(
```

Comment:
{CODENAME_TO_RIGHTS[codename] for codename in permissions if codename in CODENAME_TO_RIGHTS} 
simplify the comprehension of rights to enhance performance